### PR TITLE
.NET SDK for Azure Event Grid Preview

### PR DESCRIPTION
**Pull request by kalyanaj**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/EventGrid/EventGrid.Tests/EventGrid.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  <!--Do not remove until VS Test Tools fixes #472-->
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="SessionRecords\EventGrid.Tests.ScenarioTests.ScenarioTests\" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Update="xunit" Version="2.3.0-beta4-build3742" />
```
Comment (shahabhijeet) [inline src/SDKs/EventGrid/EventGrid.Tests/EventGrid.Tests.csproj]: @kalyanaj all these references are already added, these are redundant.
We would like to reference certain common nuget references from a common location which helps in upgrading.

File: src/SDKs/EventGrid/global.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "projects": [ "Microsoft.Azure.Management.EventGrid" ]
```
Comment (shahabhijeet) [inline src/SDKs/EventGrid/global.json]: @kalyanaj you do not need this file

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
If this is not your final PR, you don't have to open a PR.
you can iterate on your API version locally and can create nuget packages for PowerShell testing.
Once you are ready for your final PR, please open a PR and I can review it.

**Comment by shahabhijeet** [comment]
You can create local nuget packages using
msbuild build.proj /p:Scope=SDKs\EventGrid /t:CreateNugetPackage
this will create nuget package under < root >\binaries\packages

**Comment by kalyanaj** [comment]
@shahabhijeet, thanks for your earlier comments, I have resolved those issues now. I have now updated this PR based on the latest API version ("2017-09-15-preview", swagger was approved and merged this morning), and have regenerated the .NET SDK based on that version and have updated my tests as well.  So, this PR is ready now for reviewing/merging. Can you please review this and let me know if you have any additional comments?

**Comment by kalyanaj** [comment]
@shahabhijeet, for your reference, the EventGrid API pull request is https://github.com/Azure/azure-rest-api-specs/pull/1670.

File: src/SDKs/EventGrid/EventGrid.Tests/EventGrid.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    <None Update="SessionRecords\**\*.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+  <!--Do not remove until VS Test Tools fixes #472-->
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+  <ItemGroup>
```
Comment (shahabhijeet) [inline src/SDKs/EventGrid/EventGrid.Tests/EventGrid.Tests.csproj]: you don't need it if you already have what you have on line 19, please remove

File: src/SDKs/EventGrid/EventGrid.Tests/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+[assembly: AssemblyCopyright("Copyright ©  2017")]
+[assembly: AssemblyTrademark("")]
+[assembly: AssemblyCulture("")]
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// Version information for an assembly consists of the following four values:
```
Comment (shahabhijeet) [inline src/SDKs/EventGrid/EventGrid.Tests/Properties/AssemblyInfo.cs]: please remove commented code that is not needed

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/EventGrid/EventGrid.Tests/Properties/AssemblyInfo.cs (hunk 4 above)
Comment (kalyanaj) [reply src/SDKs/EventGrid/EventGrid.Tests/Properties/AssemblyInfo.cs]: Resolved.

File: src/SDKs/EventGrid/EventGrid.Tests/EventGrid.Tests.csproj (hunk 3 above)
Comment (kalyanaj) [reply src/SDKs/EventGrid/EventGrid.Tests/EventGrid.Tests.csproj]: Resolved.

**Review by shahabhijeet — APPROVED** [review]

**Comment by brandonh-msft** [comment]
From what i can tell this is an SDK for **management** and not **usage** of EventGrid. Do we have plans to create/release something like this? or is it already out and I'm just unable to discover it on nuget....

**Comment by kalyanaj** [comment]
@brandonh-msft , yes, this is the SDK for the management operations (there's no currently no SDK for _publishing_ to Event Grid and it is through the REST API calls). Yes, we do have plans for creating that SDK as well.

**Comment by brandonh-msft** [comment]
👍 
i published a stopgap [https://blogs.msdn.microsoft.com/brandonh/2017/11/30/publishing-to-an-event-grid-topic-from-net/](here) :)

**Pull request merged**